### PR TITLE
UPSTREAM: Retry service IP repair on conflict

### PR DESCRIPTION
**Pull request by smarterclayton**

### CODE REVIEW

**Comment by ncdc** [comment]
I really don't want to merge any UPSTREAM commits unless they're already merged upstream, if at all possible. Is it ok to wait for https://github.com/kubernetes/kubernetes/pull/16067 and then the equivalent to this PR upstream?

**Comment by smarterclayton** [comment]
This is now upstreamed.

**Comment by ncdc** [comment]
Update the commit and PR title to 20007?

**Comment by smarterclayton** [comment]
Commit already had it.

File: Godeps/_workspace/src/k8s.io/kubernetes/pkg/registry/service/ipallocator/controller/repair.go [hunk 1]
```diff
@@ -73,6 +75,12 @@ func (c *Repair) RunUntil(ch chan struct{}) {
 
 // RunOnce verifies the state of the cluster IP allocations and returns an error if an unrecoverable problem occurs.
 func (c *Repair) RunOnce() error {
+	return client.RetryOnConflict(client.DefaultBackoff, c.runOnce)
+}
+
+// runOnce verifies the state of the cluster IP allocations and returns an error if an unrecoverable problem occurs. It
+// will retry conflicts up to retry times.
```
Comment (ncdc) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/registry/service/ipallocator/controller/repair.go]: Update doc to match upstream change

**Comment by pmorie** [comment]
[test]

**Comment by smarterclayton** [comment]
Going to hold any merges of upstreams until the rebase lands

**Comment by smarterclayton** [comment]
UI test

```
Running "protractor:chrome" (protractor) task
Starting selenium standalone server...
[launcher] Running 1 instances of WebDriver
[launcher] Process exited with error code 1
>> /data/src/github.com/openshift/origin/assets/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/promise.js:1761
>>       throw error;
>>             ^
>> Error: Timed out waiting for the WebDriver server at http://172.18.3.126:55212/wd/hub
>>     at Error (<anonymous>)
>>     at onResponse (/data/src/github.com/openshift/origin/assets/node_modules/protractor/node_modules/selenium-webdriver/http/util.js:87:11)
>>     at /data/src/github.com/openshift/origin/assets/node_modules/protractor/node_modules/selenium-webdriver/http/util.js:42:21
>>     at /data/src/github.com/openshift/origin/assets/node_modules/protractor/node_modules/selenium-webdriver/lib/webdriver/http/http.js:96:5
>>     at ClientRequest.<anonymous> (/data/src/github.com/openshift/origin/assets/node_modules/protractor/node_modules/selenium-webdriver/http/index.js:145:7)
>>     at ClientRequest.emit (events.js:95:17)
>>     at Socket.socketErrorListener (http.js:1552:9)
>>     at Socket.emit (events.js:95:17)
>>     at net.js:441:14
>>     at process._tickCallback (node.js:442:13)
Fatal error: protractor exited with code: 8


Execution Time (2016-01-25 19:18:27 UTC)
concurrent:server  12.4s  ▇▇▇▇▇▇▇▇▇▇▇▇▇ 29%
protractor:chrome  30.2s  ▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇ 70%
Total 43.1s
```

**Comment by smarterclayton** [comment]
[test]

**Comment by ncdc** [comment]
Test timed out:
```
FAIL	github.com/openshift/origin/pkg/cmd/server/admin	60.341s
```

[test]

**Comment by smarterclayton** [comment]
This will land in the next rebase.

**Pull request closed without merging**